File: examserver/src/main/java/com/exam/services/QuestionService.java
package com.exam.services;

import com.exam.model.exam.Question;
import com.exam.model.exam.Quiz;

import java.util.Set;

public interface QuestionService {

    Question addQuestion(Question question);
    Question updateQuestion(Question question);
    Set<Question> getQuestions();
    Question getQuestion(Long questionId);
    Set<Question> getQuestionsOfQuiz(Quiz quiz);
    void deleteQuestion(Long questId);

    Question get(Long questionId);

}

File: examfront/src/app/services/question.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {
 
  constructor(private http: HttpClient) { }


  //used by admin
  public getQuestionsofQuiz(qid: any){
    return this.http.get(`${baseUrl}/question/quiz/all/${qid}`);
  }


  //used by user
  public getQuestionsofQuizForTest(qid: any){
    return this.http.get(`${baseUrl}/question/quiz/${qid}`);
  }


  public addQuestion(question:any){
    return this.http.post(`${baseUrl}/question/`,question);
  }


  //delete question
  public deleteQuestion(questionId: any){
    return this.http.delete(`${baseUrl}/question/${questionId}`);
  }

  // //eval quiz
  // public evalQuiz(questions:any){
  //   return this.http.post(`${baseUrl}/question/eval-quiz`, questions);
  // }

}
